src/shim.c

```diff
@@ -703,6 +704,9 @@ handle_proxy_message(struct cc_shim *shim)
 
 	fr = read_frame(shim);
 	if ( !fr) {
+		if (! reconnect_to_proxy(shim)) {
```

Comment:
@sboeuf, thanks for the review.

Okay. Re-connection when header length or header version is wrong doesn't make much sense. If I change for these cases instead of returning false to abort()? Would it make sense for you?